src/math/mat4.js

```diff
@@ -410,6 +410,54 @@ pc.extend(pc, (function () {
             return res.set(x, y, z);
         },
 
+         /**
+         * @function
+         * @name pc.Mat4#transformVec4
+         * @description Transforms a 4-dimensional vector by a 4x4 matrix.
+         * @param {pc.Vec4} vec The 4-dimensional vector to be transformed.
+         * @param {pc.Vec4} [res] An optional 4-dimensional vector to receive the result of the transformation.
+         * @returns {pc.Vec4} The input vector v transformed by the current instance.
+         * @example
+         * // Create a 4-dimensional vector
+         * var v = new pc.Vec4(1, 2, 3, 4);
+         *
+         * // Create a 4x4 rotation matrix
+         * var m = new pc.Mat4().setFromEulerAngles(10, 20, 30);
+         *
+         * var tv = m.transformVec4(v);

```

Comment:
Please write example with provided second argument as more recommended one to avoid allocations.